lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
> can you think of any cases where you may do something like ..
> I feel like doing the former would just be bad practice.

I'd prefer later as well, but we simply can't say for everyone, thus "whatever can happen will happen".
I think better documentation will help.